src/libstd/sys/windows/fs.rs

```diff
@@ -517,6 +517,25 @@ pub fn rmdir(p: &Path) -> io::Result<()> {
     Ok(())
 }
 
+pub fn remove_dir_all(path: &Path) -> io::Result<()> {
+    for child in try!(readdir(path)) {
+        let child = try!(child).path();
+        let stat = try!(lstat(&*child));
```

Comment:
I think this should also be able to avoid the call to `lstat` because a `DirEntry` on Windows actually has an almost complete copy of the metadata with it already.